data-access-layer/iaas/AliClient.js

```diff
@@ -0,0 +1,203 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const Storage = require('ali-oss');
+const logger = require('../../common/logger');
+const BaseCloudClient = require('./BaseCloudClient');
+
+class AliClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+    logger.debug('Looking for container ' + container);
+    return Promise.try(() => {
+      return this.storage.listBuckets({
+        prefix: container
+      })
+        .then(buckets => {
+          if (buckets.buckets === null) {
+            logger.error('Bucket ' + container + ' does not exists');
+          } else if (buckets.buckets.length == 1) {
+            return buckets.buckets;
+            logger.info('Bucket ' + container + ' exists');
+          } else {
+            logger.error('More than 1 Buckets with prefix ' + container + ' exists');
+          }
+        })
+        .catch(err => {
+          logger.error('Bucket ' + container + ' does not exists');
+        });
+    });
+  }
+
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    return this.storage
+      .useBucket(container)
+      .list(options)
+      .then(listOfFiles => {
+        let list = listOfFiles.objects;
+        let isTruncated = listOfFiles.isTruncated;
+        let marker = listOfFiles.nextMarker;
+        const files = [];
+        _.each(list, file => files.push(_
+          .chain(file)
+          .pick('name', 'lastModified')
+          .set('isTruncated', isTruncated)
+          .set('marker', marker)
+          .value()
+        ));
+        return files;
+      });
+  }
+
+  listFilenames(prefix, max_keys) {
```

Comment:
This function is not required in this class. It is already there in [BackupStore.js](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/33153976cbdff27ca51dc96d513a29c8a3dd9d17/data-access-layer/iaas/BackupStore.js#L35). It was added in PoC code only to test/mimic what is there in BackupStore.

Reply:
removed! Thanks!